src/box/txn.h

```diff
@@ -557,6 +557,8 @@ struct txn {
 	 * Nesting level of space on_replace triggers for current txn.
 	 */
 	int space_on_replace_triggers_depth;
+	/** Whether the transaction is linearizable. */
+	bool is_linearizable;
```

Comment:
AFAICS, the flags are supposed to be stored in `txn.flags` and should all be listed in `enum txn_flag`.

Reply:
I didn't put `IS_LINEARIZABLE` to flags initially, because I thought `flags` hold technical details, like IS_ABORTED, or IS_CONFLICTED or CAN_YIELD rather than txn mode or txn isolation.

But I see that `TXN_WAIT_SYNC` is in flags, so I'll make `linearizable` a flag as well:

<details><summary>diff</summary>

```diff
diff --git a/src/box/txn.c b/src/box/txn.c
index 5a70c7394..5c8024a74 100644
--- a/src/box/txn.c
+++ b/src/box/txn.c
@@ -553,7 +553,6 @@ txn_begin(void)
        txn->fiber = NULL;
        txn->timeout = TIMEOUT_INFINITY;
        txn->rollback_timer = NULL;
-       txn->is_linearizable = false;
        fiber_set_txn(fiber(), txn);
        trigger_create(&txn->fiber_on_yield, txn_on_yield, NULL, NULL);
        trigger_add(&fiber()->on_yield, &txn->fiber_on_yield);
@@ -608,7 +607,7 @@ txn_begin_stmt(struct txn *txn, struct space *space, uint16_t type)
                txn_set_flags(txn, TXN_IS_CONFLICTED);
        }
 
-       if (space != NULL && txn->is_linearizable &&
+       if (space != NULL && txn_has_flag(txn, TXN_IS_LINEARIZABLE) &&
            engine_check_space_linearizability(space->engine, space) != 0) {
                return -1;
        }
@@ -1422,7 +1421,7 @@ box_txn_make_linearizable(void)
        }
        if (box_wait_linearization_point(txn->timeout) != 0)
                return -1;
-       txn->is_linearizable = true;
+       txn_set_flags(txn, TXN_IS_LINEARIZABLE);
        return 0;
 }
 
diff --git a/src/box/txn.h b/src/box/txn.h
index 1d8701ee2..a8a7e4501 100644
--- a/src/box/txn.h
+++ b/src/box/txn.h
@@ -101,6 +101,11 @@ enum txn_flag {
         * rolled back at commit.
         */
        TXN_IS_ABORTED_BY_TIMEOUT = 0x100,
+       /**
+        * Transaction is linearizable, so it can only touch certain types of
+        * spaces (synchronous ones, for example).
+        */
+       TXN_IS_LINEARIZABLE = 0x200,
 };
 
 enum {
@@ -557,8 +562,6 @@ struct txn {
         * Nesting level of space on_replace triggers for current txn.
         */
        int space_on_replace_triggers_depth;
-       /** Whether the transaction is linearizable. */
-       bool is_linearizable;
 };
 
 static inline bool
@@ -798,7 +801,7 @@ txn_begin_ro_stmt(struct space *space, struct txn **txn,
                        diag_set(ClientError, ER_TRANSACTION_CONFLICT);
                        return -1;
                }
-               if ((*txn)->is_linearizable &&
+               if (txn_has_flag(*txn, TXN_IS_LINEARIZABLE) &&
                    engine_check_space_linearizability(space->engine,
                                                       space) != 0) {
                        return -1;

```

</details>